### ISSUE TITLE
Broken image calculation on specific image formats

### ISSUE THREAD

**Issue by christianbader** [sulu/sulu#2447]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | Specific version or SHA of a commit
| Browser Version | Browser name and version

#### Actual Behavior

Broken links for some images with a specific (huge width, small height) proportion. (See Attached files b123750-1.jpg)

Exception -> "Length of either side cannot be 0 or negative, current size is 12x0"

#### Expected Behavior

Correct scaling of the appropriate image.

#### Attached files

Broken:
![b123750-1](https://cloud.githubusercontent.com/assets/96615/15777864/d3702d7c-2992-11e6-985a-fb2e7cfdce86.jpg)

Wrong scale:
![b123302_hb-1](https://cloud.githubusercontent.com/assets/96615/15777875/ea8be320-2992-11e6-8d94-abcaf9f6edef.jpg)

#### Possible Solutions

If the scaled height is < 1px set it to 1px.

**Comment by wachterjohannes** [comment]
@christianbader i hope you still have this images (: but you was scaled here. please send me them with slack and paste the images-formats config here.

**Comment by christianbader** [comment]
```
<?xml version="2.0" encoding="UTF-8"?>
<formats xmlns="http://schemas.sulu.io/media/formats"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://schemas.sulu.io/media/formats http://schemas.sulu.io/media/formats-1.0.xsd">
    <format>
        <name>70x70</name>
        <commands>
            <command>
                <action>resize</action>
                <parameters>
                    <parameter name="x">70</parameter>
                    <parameter name="y">70</parameter>
                </parameters>
            </command>
        </commands>
    </format>

    <format>
        <name>square140</name>
        <commands>
            <command>
                <action>scale</action>
                <parameters>
                    <parameter name="mode">inset</parameter>
                    <parameter name="x">140</parameter>
                    <parameter name="y">140</parameter>
                </parameters>
            </command>
        </commands>
    </format>

    <format>
        <name>square740</name>
        <commands>
            <command>
                <action>scale</action>
                <parameters>
                    <parameter name="mode">inset</parameter>
                    <parameter name="x">740</parameter>
                    <parameter name="y">740</parameter>
                </parameters>
            </command>
        </commands>
    </format>

    <format>
        <name>770x350</name>
        <commands>
            <command>
                <action>scale</action>
                <parameters>
                    <parameter name="x">770</parameter>
                    <parameter name="y">350</parameter>
                </parameters>
            </command>
        </commands>
    </format>

    <format>
        <name>770x485</name>
        <commands>
            <command>
                <action>scale</action>
                <parameters>
                    <parameter name="x">770</parameter>
                    <parameter name="y">485</parameter>
                </parameters>
            </command>
        </commands>
    </format>

    <format>
        <name>870x130</name>
        <commands>
            <command>
                <action>scale</action>
                <parameters>
                    <parameter name="x">870</parameter>
                    <parameter name="y">130</parameter>
                </parameters>
            </command>
        </commands>
    </format>

    <format>
        <name>770x115</name>
        <commands>
            <command>
                <action>scale</action>
                <parameters>
                    <parameter name="x">770</parameter>
                    <parameter name="y">115</parameter>
                </parameters>
            </command>
        </commands>
    </format>

    <format>
        <name>270x550</name>
        <commands>
            <command>
                <action>scale</action>
                <parameters>
                    <parameter name="x">270</parameter>
                    <parameter name="y">550</parameter>
                </parameters>
            </command>
        </commands>
    </format>

    <format>
        <name>270x270</name>
        <commands>
            <command>
                <action>scale</action>
                <parameters>
                    <parameter name="x">270</parameter>
                    <parameter name="y">270</parameter>
                </parameters>
            </command>
        </commands>
    </format>
</formats>
```

**Comment by wachterjohannes** [comment]
the wrong scale can be fixed by adding the parameter `forceRatio = false` then the images will be also upscalled.

**Comment by wachterjohannes** [comment]
or only type in the with you want (leave y empty).

**Comment by alexander-schranz** [comment]
the bug is when the calculation is 0 so this need to be checked in the scale command that it never could be 0.

**Comment by wachterjohannes** [comment]
the problem is that we dont say we want 6x0 this will be calculated in the imagine library.

**Comment by danrot** [comment]
Fixed in #2454
